app/src/androidTest/java/org/kiwix/kiwixmobile/KiwixMockServer.kt

```diff
@@ -0,0 +1,139 @@
+package org.kiwix.kiwixmobile
+
+import okhttp3.mockwebserver.Dispatcher
+import okhttp3.mockwebserver.MockResponse
+import okhttp3.mockwebserver.MockWebServer
+import okhttp3.mockwebserver.RecordedRequest
+import org.kiwix.kiwixmobile.library.entity.LibraryNetworkEntity
+import org.kiwix.kiwixmobile.library.entity.LibraryNetworkEntity.Book
+import org.kiwix.kiwixmobile.library.entity.MetaLinkNetworkEntity
+import org.kiwix.kiwixmobile.library.entity.MetaLinkNetworkEntity.FileElement
+import org.kiwix.kiwixmobile.library.entity.MetaLinkNetworkEntity.Pieces
+import org.kiwix.kiwixmobile.library.entity.MetaLinkNetworkEntity.Url
+import org.simpleframework.xml.core.Persister
+import java.io.StringWriter
+import java.util.LinkedList
+import java.util.concurrent.TimeUnit
+
+/*
+ * Kiwix Android
+ * Copyright (C) 2018  Kiwix <android.kiwix.org>
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ */
+class KiwixMockServer {
+  val mockWebServer = MockWebServer().apply {
+    start(8080)
+  }
+
+  fun stop() {
+    mockWebServer.shutdown()
+  }
+
+  fun enqueue(libraryNetworkEntity: LibraryNetworkEntity) {
+    mockWebServer.enqueue(MockResponse().apply {
+      setResponseCode(200)
```

Comment:
Should we use [HTTP_OK](https://developer.android.com/reference/java/net/HttpURLConnection.html#HTTP_OK) here?